### ISSUE TITLE
Overlap detected after sprite removal

### ISSUE THREAD

**Issue by ghost** [molleindustria/p5.play#146]
It seems as though in certain situations, overlaps between removed objects still get detected. 

```
var player;
var pickup;
var enemies;
var score = 0;

function setup() {
	createCanvas(600, 400);

	enemies = new Group();

	for (var i = 0; i < 5; i++) {
		var enemy = createSprite(100 + 100 * i, 0, 50, 50);
		enemy.shapeColor = color(0);
		enemy.velocity.y = 2;
		enemies.add(enemy);
	}

	player = createSprite(300, 200, 20, 20);
	player.shapeColor = color(255);

	pickup = createSprite(random(0, width), random(0, height), 20, 20);
	pickup.shapeColor = color(255, 0, 0);
}

function draw() {
	background(200);

	// overlap detected even after it's removed
	if (player.overlap(pickup)) {
		pickup.remove();
		score += 20;
	}
	
	textSize(36);
	text(score, 20, 40);

	if (player.overlap(enemies)) {
		player.remove();
	}

	player.position.x = mouseX;
	player.position.y = mouseY;

	drawSprites();
}
```

In the following example, score gets incremented after the pickup is removed.

Seems like a missing ```!sprite.removed``` in collision detection.

**Comment by quinton-ashley** [comment]
This is still an issue! I will fix this.

**Comment by quinton-ashley** [comment]
Fixed in version 2.2.2

Pretty easy fix, just had to not check the sprite for collisions if it was removed.
